### PR TITLE
restore @ibab's home_handler 

### PR DESCRIPTION
**Pull request by anaderi**

### CODE REVIEW

**Comment by betatim** [comment]
It is good to see things happening quickly. Gathering momentum is good. However right now it feels like driving at 200kph in a 20yr old car on a windy mountain road. Too many changes with too little focus on the long term maintainability. Rome wasn't built in a day, so we should take our time too.

It is much more important to have high quality (tested, documented and reviewed) code than high momentum. In the long term we will suffer from making changes quickly instead of slowing down, making small well defined (problem -> solution) changes, letting people look at them, think, comment, rework them, etc. Neither of us is perfect so we will make mistakes, and the only way to catch those is to have someone else take a look. It is also important to let everyone take part, otherwise we alienate them. Giving people a chance to take part means waiting for people to have time to look and think. Yes it is frustrating if you are on a roll and you want to see your changes out there, but we are a team of volunteers who lead busy lives in different timezones. We have to respect that.

The default mode of operation should be: point out a problem, discuss what to do, make a PR, come back the next day, see if someone has looked, commented and so on. There are no deadlines, funding or status updates to management. The only pressure we have with this project is to get things right and build a high quality product.

So with this in mind:

:cry: Where are the unit and frontend test for this? In #67 we discussed maintaining @ibab's authorship of the work when cherry picking the changes, why didn't this happen? We also discussed in gitter to split this into three separate PRs, why not do that?

**Comment by anaderi** [comment]
You are right, I agree in general, but there are also several things to consider (I'm not trying to justify myself, just to give some reasons):
- @ibab 's commit has been around for some time without tests -- we've agreed to have it in master
- then we've deleted it since travis was unhappy 
- later on we've figured that travis unhappiness has nothing to do with those commits, so we have to get it back
- as I put it back, I mention @ibab in PR (are there other ways of doing so?)
- this commit doesn't split into 3 pieces naturally (or it is not obvious for me)
- we are still in beta stage, implying users agree to risk stability vs functionality
- regarding tests - I have no clear idea how can I test fork/push automatically since it involves github. (test for repourl is not valid as well, since this functionality works only with githubauthenticator) thoughts? 

in other words, I'm not sure we are trying to build Rome in one night (or riding brakeless car down the hill), I just try to collect some feedback that would help making the product a bit better. 

As another consideration I'll try to create another instance for quality assurance, that would allow for less tested deployments.

**Comment by betatim** [comment]
I agree on the beta thing. I don't care so much about users but about us and how easy it is to maintain the project. For this we need to make sure everyone involved understands why the code is the way it is. Understanding what code does is much easier than why it does it. The why is far more important though. I'd like us to develop a model of development that increases the amount of understanding-why as much as possible.

Split: we could have had one PR that just adds "we detected there are changes" once we had that working, a second PR that builds on it "should we push this to :octocat: for you?"

Testing: we will have to mock things and it will be a bit tricky, this is why we should go slow and discuss it.  I'd go as far as saying that we should not add a feature if we can not think of how to test it. In the long term it is better for the project not to have that feature (until we can work out how to test it). We did not insist on tests since day one, so at some point there will be an arbitrary moment after which changes will only be accepted if they have tests. It will feel unfair ("if only you had merged this 2hrs ago I wouldn't need to add tests!!?!") but I don't see how else we can make the transition. If writing tests feels like a burden/not a fun thing to do maybe that person should find a different project to contribute to :trollface: 

In general there will be many things were none of us know exactly how to do it. Often one of us will have an idea though, then we discuss it and arrive at a solution that is better than what any individual could have cooked up.

Credit: credit is the only thing the project can give back to the contributors. If it means going an extra mile, I think we should do it. In this case the way would have been to cherry pick the commits, and add one commit that fixes the conflicts.

**Comment by anaderi** [comment]
General: Well, I guess it is something personal for any maintainer how willing is he to understand all the details of all the code and all the underlying libraries. In my case since I'm using this project for teaching, it is more important to have it working (as I told you openml is willing to integrate `integrate`/`promote` everware so there might impose some workability constraints as well). Or maybe this interpretabilty vs functionality balance refers to something  more personal like extraversion/introversion? 
Anyway good to know your position on that. Just wanted to let you know that mine is a bit different. 
I'd be very keen to see what kind of development/maintenance model would work in that case. I'm not sure the best one would be that implies vetoing PRs by a single maintainer. Probably this one would be good place to follow the discussion: https://github.com/everware/everware/pull/89

Split, Credit: I'd say splitting code in pieces is impractical since the code that does it basically is already there. I don't mind at all if @ibab removes it again (or I can do it myself) and commits it back to keep the authorship. as I said above I'm more in favor of functionality. (also, if we care about credit so much, maybe we should be more careful about just removing whole bunch of code in a single PR?)

Testing: I totally agree. the only thing is that we've already agreed on having the code to be in master. it is WIP, so let's discuss github mocking: https://github.com/everware/everware/issues/90

**Comment by betatim** [comment]
Agree that removing stuff is not ideal and I might have been over-reverting. There was for sure some red mist as it was frustrating to see #72 not pass the tests because `master` itself was red and then spending 6hrs getting back to having `master` build just so I could get a change to `.gitignore` merged. I'd probably advocate a fairly ruthless 'revert first, ask questions later' policy in cases where a merge breaks master. In this case our tests weren't yet good enough because they sometimes failed "at random"...overall I think it was just not an ideal situation.

More comments on general, testing and splitting in #89
